rat/codegen_backends.py

```diff
@@ -33,187 +33,157 @@ def __init__(self, symbol_table: "compiler_rewrite.SymbolTable", primary_variabl
-        self.expression_string += "jax.numpy.exp("
-        exp_node.subexpr.accept(self, *args, **kwargs)
-        self.expression_string += ")"
-
-    def visit_Abs(self, abs_node: ast.Abs, *args, **kwargs):
-        self.expression_string += "jax.numpy.abs("
-        abs_node.subexpr.accept(self, *args, **kwargs)
-        self.expression_string += ")"
-
-    def visit_Floor(self, floor_node: ast.Floor, *args, **kwargs):
-        self.expression_string += "jax.numpy.floor("
-        floor_node.subexpr.accept(self, *args, **kwargs)
-        self.expression_string += ")"
-
-    def visit_Ceil(self, ceil_node: ast.Ceil, *args, **kwargs):
-        self.expression_string += "jax.numpy.ceil("
-        ceil_node.subexpr.accept(self, *args, **kwargs)
-        self.expression_string += ")"
-
-    def visit_Round(self, round_node: ast.Round, *args, **kwargs):
-        self.expression_string += "jax.numpy.round("
-        round_node.subexpr.accept(self, *args, **kwargs)
-        self.expression_string += ")"
-
-    def visit_Sin(self, sin_node: ast.Sin, *args, **kwargs):
-        self.expression_string += "jax.numpy.sin("
-        sin_node.subexpr.accept(self, *args, **kwargs)
-        self.expression_string += ")"
-
-    def visit_Cos(self, cos_node: ast.Cos, *args, **kwargs):
-        self.expression_string += "jax.numpy.cos("
-        cos_node.subexpr.accept(self, *args, **kwargs)
-        self.expression_string += ")"
-
-    def visit_Tan(self, tan_node: ast.Tan, *args, **kwargs):
-        self.expression_string += "jax.numpy.tan("
-        tan_node.subexpr.accept(self, *args, **kwargs)
-        self.expression_string += ")"
-
-    def visit_Arcsin(self, arcsin_node: ast.Arcsin, *args, **kwargs):
-        self.expression_string += "jax.numpy.arcsin("
-        arcsin_node.subexpr.accept(self, *args, **kwargs)
-        self.expression_string += ")"
-
-    def visit_Arccos(self, arccos_node: ast.Arccos, *args, **kwargs):
-        self.expression_string += "jax.numpy.arccos("
-        arccos_node.subexpr.accept(self, *args, **kwargs)
-        self.expression_string += ")"
-
-    def visit_Arctan(self, arctan_node: ast.Arctan, *args, **kwargs):
-        self.expression_string += "jax.numpy.arctan("
-        arctan_node.subexpr.accept(self, *args, **kwargs)
-        self.expression_string += ")"
-
-    def visit_Logit(self, logit_node: ast.Logit, *args, **kwargs):
-        self.expression_string += "jax.scipy.special.logit("
-        logit_node.subexpr.accept(self, *args, **kwargs)
-        self.expression_string += ")"
-
-    def visit_InverseLogit(self, invlogit_node: ast.InverseLogit, *args, **kwargs):
-        self.expression_string += "jax.scipy.special.expit("
-        invlogit_node.subexpr.accept(self, *args, **kwargs)
-        self.expression_string += ")"
-
-
-class EvaluateDensityVisitor(BaseVisitor):
+    def generate(self, ast_node: ast.Expr):
+        match ast_node:
+            case ast.IntegerConstant():
```

Comment:
Can you check if your IDE picks up the node attributes(`subexpr, value, lhs, etc.`) within each case statement? Pycharm still identifies them as `Expr`, which is making me uncomfortable.

Reply:
Yeah vscode is catching the types correctly. Maybe cause match is a 3.10 feature it'll just take a second till all the stuff is supported.

Reply:
Thanks!